fast64_internal/render_settings.py

```diff
@@ -1,253 +1,265 @@
-        default=(1, 1, 1, 1),
-        update=on_update_render_preview_nodes,
-    )
-    ambientColor: bpy.props.FloatVectorProperty(
-        name="Ambient Light",
-        subtype="COLOR",
-        size=4,
-        min=0,
-        max=1,
-        default=(0.5, 0.5, 0.5, 1),
-        update=on_update_render_preview_nodes,
-    )
-    lightColor: bpy.props.FloatVectorProperty(
-        name="Light Color",
-        subtype="COLOR",
-        size=4,
-        min=0,
-        max=1,
-        default=(1, 1, 1, 1),
-        update=on_update_render_preview_nodes,
-    )
-    lightDirection: bpy.props.FloatVectorProperty(
-        name="Light Direction",
-        subtype="DIRECTION",
-        size=3,
-        min=-1,
-        max=1,
-        default=mathutils.Vector((0.5, 0.5, 1)).normalized(),  # pre normalized
-        update=on_update_render_preview_nodes,
-    )
-    useWorldSpaceLighting: bpy.props.BoolProperty(
-        name="Use World Space Lighting", default=True, update=on_update_render_settings
-    )
-    # Fog Preview is int because values reflect F3D values
-    fogPreviewPosition: bpy.props.IntVectorProperty(
-        name="Fog Position", size=2, min=0, max=0x7FFFFFFF, default=(985, 1000), update=on_update_render_preview_nodes
-    )
-    # Clipping planes are float because values reflect F3D values
-    clippingPlanes: bpy.props.FloatVectorProperty(
-        name="Clipping Planes", size=2, min=0, default=(100, 30000), update=on_update_render_preview_nodes
-    )
-    useObjectRenderPreview: bpy.props.BoolProperty(
-        name="Use Object Preview", default=True, update=on_update_render_settings
-    )
-    # SM64
-    sm64Area: bpy.props.PointerProperty(
-        name="Area Object", type=bpy.types.Object, update=on_update_sm64_render_settings, poll=poll_sm64_area
-    )
-    # OOT
-    ootSceneObject: bpy.props.PointerProperty(
-        name="Scene Object", type=bpy.types.Object, update=on_update_oot_render_settings, poll=poll_oot_scene
-    )
-    ootSceneHeader: bpy.props.IntProperty(
-        name="Header/Setup",
-        description="Scene header / setup to use lighting data from",
-        min=0, soft_max=10, default=0,
-        update=on_update_oot_render_settings,
-    )
-    ootForceTimeOfDay: bpy.props.BoolProperty(
-        name="Force Time of Day",
-        description="Interpolate between four lights based on the time",
-        default=False,
-        update=on_update_oot_render_settings,
-    )
-    ootLightIdx: bpy.props.IntProperty(
-        name="Light Index",
-        min=0, soft_max=10, default=0,
-        update=on_update_oot_render_settings,
-    )
-    ootTime: bpy.props.FloatProperty(
-        name="Time of Day (Hours)",
-        description="Time of day to emulate lighting conditions at, in hours",
-        min=0.0, max=23.99, default = 10.0, precision=2, subtype="TIME", unit="TIME",
-        update=on_update_oot_render_settings,
-    )
-    
+import bpy
+import mathutils
+import math
+from .utility import *
```

Comment:
someday I need to get rid of the remaining ``from x import *`` 🙂

Reply:
pepeCrusade